udata/core/dataset/apiv2.py

```diff
@@ -404,6 +404,58 @@ def get(self, dataset):
         }
 
 
+@ns.route("/<dataset:dataset>/schemas/", endpoint="dataset_schemas", doc=common_doc)
+@apiv2.response(404, "Dataset not found")
+@apiv2.response(410, "Dataset has been deleted")
+class DatasetSchemasAPI(API):
+    @apiv2.doc("get_dataset_schemas")
+    @apiv2.marshal_with(schema_fields)
+    def get(self, dataset):
+        """Get a dataset schemas given its identifier"""
+        if dataset.deleted and not DatasetEditPermission(dataset).can():
```

Comment:
Do we want to abort for a private dataset too ?

Reply:
Yes, was waiting for the other PR to be merged to still the code :-) Done in 7aaca0ff1